metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -454,17 +464,23 @@ class TableauConfig(
     def projects_backward_compatibility(cls, values: Dict) -> Dict:
         projects = values.get("projects")
         project_pattern = values.get("project_pattern")
-        if project_pattern is None and projects:
+        project_path_pattern = values.get("project_path_pattern")
+        if project_pattern is None and project_path_pattern is None and projects:
             logger.warning(
-                "project_pattern is not set but projects is set. projects is deprecated, please use "
-                "project_pattern instead."
+                "projects is deprecated, please use " "project_path_pattern instead."
             )
             logger.info("Initializing project_pattern from projects")
             values["project_pattern"] = AllowDenyPattern(
```

Comment:
@sid-acryl I now deprecated `project_pattern` but I wanted to point out that for now this validation rule still initializes `project_pattern` when `projects` is set. Ideally, we would remove the old `projects` completely now but I tried it and it would require quite some refactoring in the tests.